### DISCUSSION TITLE
[HW5] Submission keep failing

### DISCUSSION THREAD

**Discussion by Akais24** [prosyslab-classroom/cs424-program-reasoning#452] [Q&A] [answered]
### Your Name

황재영

### Question

HW5 코드 제출이 지속적으로 실패하여 문의드리려고 합니다
assert나 print 등, 의심되는 포인트를 제거해보았지만 해결되지 않았습니다

https://www.gradescope.com/courses/840493/assignments/4937517/submissions/292928075

--- 

I'm writing to inquire about the continuous failure of my HW5 code submission.
Despite removing suspected points like assert statements and print functions, the issue remains unresolved.

https://www.gradescope.com/courses/840493/assignments/4937517/submissions/292928075

**Comment by spearo2** [comment]
안녕하세요.
확인 결과 제출하신 재영님의 합성기가 gradescope에서 할당하는 시스템 자원을 초과 사용하여 `SIGKILL` 시그널을 받아 프로세스가 죽는것을 확인하였습니다.
또한, 다른 학생분들의 제출물 및 지난 학기의 챔피언들의 제출물에서는 해당 문제가 없기 때문에 때문에 시스템 자원이 합성기 실행에 부족하지 않은것을 확인하였습니다. 
혹시 `test`디렉토리에 있는 예제를 통해 문제없이 합성기가 작동하는지 확인해 보셨나요?

감사합니다.

---
Hello,
It seems your process exceeded the limit on the number of system resources you can use.
However, we could not find the same problems in other students' submissions, including the previous year's champions, indicating that the resources are enough for the given assignment.
Have you checked if your synthesizer works fine with the examples in `test` directory on your machine?

Thank you

**Comment by Akais24** [comment]
네
제출에 사용한 코드를 로컬 환경의 도커 컨테이너 위에서 빌드하여 실행했을 때는
test 디렉토리 하위에 있는 2개의 테스트케이스와, mytest 디렉토리에 생성한 5개의 테스트케이스는 모두 10초 이내로 실행됨으로 확인하였습니다

다만 로컬 환경에서는 특별히 memory에 제약을 두지 않았는데, 이를 고려해야 다시 구현해보도록 하겠습니다

---

Yes
When building and running the submitted code on a Docker container in the local environment,
I confirmed that all 2 test cases in the test directory and 5 test cases created in the mytest directory were executed within 10 second.

However, since no specific memory constraints were set in the local environment,
I will reconsider and re-implement the code with memory considerations in mind.

**Comment by Akais24** [comment, marked as answer]
구현 방식을 바꾸어 해결하였습니다

---

I have solved the issue by changing the implementation method.

**Reply by spearo2** [reply]
Duet을 이기셨군요!

---

You beated Duet!
